src/components/DeckMechanisms/index.js

```diff
@@ -351,6 +364,43 @@ export default class DeckMechanisms extends React.Component {
       deck: this.getIncreasedDeckWeight({ state, reset }),
     }))
 
+  resolveManaRNG = state => {
+    switch (this.state.RNG) {
+      case 'UNFRIENDLY': {
+        state.specifics.activeFrozenCores = 0
+        state.specifics.activeDawnsparks = 0
+        break
+      }
+
+      case 'REGULAR': {
+        const { activeFrozenCores, activeDawnsparks } = state.specifics
+
+        state.specifics.activeFrozenCores = getBinomialRandomVariableResult(
+          activeFrozenCores,
+          FROZEN_CORE_STAYS
+        )
+        state.specifics.activeDawnsparks = getBinomialRandomVariableResult(
+          activeDawnsparks,
+          DAWNSPARKS_STAYS
+        )
+        break
+      }
+
+      default:
+        break
+    }
+
+    const { activeDawnsparks } = state.specifics
+
+    const dawnsparksManagingToGiveMana =
+      this.state.RNG === 'REGULAR'
+        ? getBinomialRandomVariableResult(activeDawnsparks, DAWNSPARKS_HITS)
```

Comment:
I feel like this should be done in switch; that’s why we have the switch in the first place.

```js
case 'REGULAR': {
  // …
  state.mana += getBinomialRandomVariableResult(activeDawnsparks, DAWNSPARKS_HITS) * 4
  // …
  break
}

Reply:
I wanted to add mana from FC and Dawnsparks in the same place for clarity, but I'll move it